flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/AsyncDataStream.scala

```diff
@@ -71,6 +71,9 @@ object AsyncDataStream {
       override def asyncInvoke(input: IN, resultFuture: JavaResultFuture[OUT]): Unit = {
         asyncFunction.asyncInvoke(input, new JavaResultFutureWrapper(resultFuture))
       }
+      override def timeout(input: IN, resultFuture: JavaResultFuture[OUT]): Unit = {
```

Comment:
Are those changes in `AsyncDataStream.scala` somewhere covered by the tests? If you comment out their method bodies, does at least one test fails?

Reply:
I haven't found any tests for `AsyncDataStream.scala` or `AsyncFunction.scala`, I am not sure whether it is missing or unnecessary. What do you think?

Reply:
You are right, there are none :( They are definitely missing, however simple this code is. I know this is painful but still I would insist on adding at least some rudimentary test coverage for the new code. As it is now, if someone would change the below invocation to `asyncFunction.timeout(null, null)` we wouldn't have known. 

Let's try to improve the situation here a little bit. The new test doesn't have to duplicate the test cases from `AsyncWaitOperatorTest.java`, but some most basic one that checks that expected arguments are being passed will suffice.

Reply:
I have added a test for `AsyncDataStream.scala`, plz review and see if it is sufficient :)